### ISSUE TITLE
Running the project

### ISSUE THREAD

**Issue by faiyaz26** [iceboy233/windows-container#1]
How to run the project ? Can you give me a brief about the project ? Or can you release a compiled version ?

**Comment by iceboy233** [comment]
This project is currently unfinished, and cannot run yet.
You can download binaries at https://code.google.com/p/winjudge/downloads/list, which is for running "judges" in a safe environment.
It has similar implementation but different (specific) purpose.

This project is like a container, where you can run unsafe binary code under the Windows environment without a virtual machine.
The main effort is to move the implementation from the "winjudge" project and make it a general purpose container library.
The native interface of this library should be C or C++, and it will have a command line interface where you can launch as "winc untrusted_binary args...", and it will also have some scripting language bindings (perhaps Node.js will be the first).

I have no time for this project in these months, and I will hopefully restart this project in the next year.
Will close this issue when the project is runnable.

**Comment by iceboy233** [comment]
After reading code of the [Chromium Sandbox](http://www.chromium.org/developers/design-documents/sandbox) project, I found it essential to rewrite the whole thing and created a new branch `master2`.

The original [winjudge](//code.google.com/p/winjudge/) and the Chromium Sandbox are largely based on the same approach, which uses a restricted token + a separated desktop + a job object. However
- The Chromium Sandbox has a stricter (yet configurable) bound on the target process, but causes a limitation where the target process _must_ be statically linked to the sandbox library. This is not we want: as the Online Judge use case, we want to be compatible to every kind of match rules, which may require the user to write the entry (main) function in various programming languages.
- The winjudge project initially used a separate user logon to create the restricted token. This is unnecessary (but useful), so we will make it available and optional in this project.

The winjudge project had a long history where we decided to support Windows 2000/XP/2003 etc. In this rewrite we absolutely want to drop support for 2K, therefore we can use the API-builtin [timer queue](//msdn.microsoft.com/en-us/library/windows/desktop/ms686796.aspx). If we drop support for Windows XP/2003, we could use [selective handle inheritance](//msdn.microsoft.com/en-us/library/windows/desktop/ms686880.aspx), [Windows integrity mechanism](//msdn.microsoft.com/en-us/library/bb625964.aspx) and the [safer API](//msdn.microsoft.com/en-us/library/windows/desktop/ms722422.aspx) which is not safe. Dropping support also means avoid the testing burden under these versions of Windows.

We may also want more functionality, such as time and memory limit, process state monitor and injection. But at this point I'm not sure if it's appropriate to put them in this layer.
In this rewrite, I want to avoid using exceptions and STL components which may generate exceptions according to the [Google C++ style guide](//google-styleguide.googlecode.com/svn/trunk/cppguide.html), which I found suitable for this level of size of project. I also want to revert back to the VS default directory structure.
